src/components/SignupForm/SignupForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Button } from '@material-ui/core';
+import SignUpFormDataInterface from './interfaces/SignupFormDataInterface';
+import ErrorMessageVisibleInterface from './interfaces/ErrorMessageVisibleInterface';
+import useSignupFormValidation from './useSignupFormValidation';
+import useFormFieldsConfig from './useFormFieldsConfig';
+import FormField from './FormField';
+import { auth } from '../../firebase';
+
+const SignupForm: React.FC = () => {
+  const initialFormData: SignUpFormDataInterface = {
+    name: '',
+    password: '',
+    repeatPassword: '',
+    email: ''
+  };
+
+  const initialErrorMessageVisible: ErrorMessageVisibleInterface = {
+    name: false,
+    email: false,
+    password: false,
+    repeatPassword: false
+  };
+
+  const [formData, setFormData] = useState(initialFormData);
+  const [errorMessageVisible, setErrorMessageVisible] = useState(initialErrorMessageVisible);
```

Comment:
please use for booleans notation with "is" prefix like isErrorMessageVisible
you could add this rule "react/boolean-prop-naming": ["error", { "rule": "^(is)[A-Z]([A-Za-z0-9]?)+" }]

Reply:
"you could add this rule "react/boolean-prop-naming": ["error", { "rule": "^(is)A-Z+" }]" - still not applied to eslint